lib/vim-plugin.nix

```diff
@@ -1,5 +1,34 @@
 { lib }:
+let
+  settingsOptionDescription =
+    { name, globalPrefix }:
+    ''
+      The configuration options for **${name}** without the `${globalPrefix}` prefix.
+
+      For example, the following settings are equivialent to these `:setglobal` commands:
+      - `foo_bar = 1` -> `:setglobal ${globalPrefix}foo_bar=1`
+      - `hello = "world"` -> `:setglobal ${globalPrefix}hello="world"`
+      - `some_toggle = true` -> `:setglobal ${globalPrefix}some_toggle`
+      - `other_toggle = false` -> `:setglobal no${globalPrefix}other_toggle`
+    '';
+
+  mkSettingsOption =
+    {
+      options ? { },
+      example ? { },
+      name,
+      globalPrefix,
+    }:
+    lib.nixvim.mkSettingsOption {
+      inherit options example;
+      description = settingsOptionDescription { inherit name globalPrefix; };
+    };
+
+  processPrefixedGlobals = prefix: lib.mapAttrs' (n: lib.nameValuePair (prefix + n));
+in
 {
+  inherit settingsOptionDescription mkSettingsOption processPrefixedGlobals;
```

Comment:
Curious why you used a `let-in` block here? Wouldn't it be neater to define all public functions in the output attrset, and then if need access them via the `lib` input argument?

Reply:
I wanted to avoid using `rec` but was not too sure. What do you prefer ?

Reply:
Not a fan of using `rec`, as it'll block us from ever making the `lib` properly extendible/overridable. We have recursive access to `lib.nixvim`. But I guess it's fine for now.

Reply:
That's why I went with `let`-`in` in the first place. Do you want me to revert to that ?

Reply:
let-in is also not overridable. So in that scenario its just as bad.

What I would suggest is `inherit (lib.nixvim) settingsOptionDescription mkSettingsOption processPrefixedGlobals`, but as I say. Not currently a big deal.